src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,378 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Improve\Design;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\MailTemplate\Command\GenerateThemeMailsCommand;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Language\LanguageInterface;
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\Layout\LayoutInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateRendererInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeCatalogInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeInterface;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Form\Admin\Improve\Design\MailTheme\GenerateMailsType;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Class MailThemeController manages mail theme generation, you can define the shop
+ * mail theme, and regenerate mail in a specific language.
+ *
+ * Accessible via "Design > Mail Theme"
+ */
+class MailThemeController extends FrameworkBundleAdminController
+{
+    /**
+     * Show mail theme settings and generation page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class);
+        /** @var ThemeCatalogInterface $themeCatalog */
+        $themeCatalog = $this->get('prestashop.core.mail_template.theme_catalog');
+        $mailThemes = $themeCatalog->listThemes();
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/MailTheme/index.html.twig', [
+            'layoutHeaderToolbarBtn' => [],
+            'layoutTitle' => $this->trans('Mail Theme', 'Admin.Navigation.Menu'),
```

Comment:
Is it a menu entry or a tab? Perhaps it will be better to write _E-mail Theme_ or _Email Theme_?

Reply:
Why not _Email Themes_ in _Admin.Design.Feature_ like in line 31 of .../Resources/views/Admin/Improve/Design/MailTheme/Blocks/list_mail_themes.html.twig?

Reply:
Alright!

Reply:
Actually `Email Theme` seems more logical, as other menu items use singular as well (`Theme & Logo`, `Theme Catalog`, ...)
But the `Email Themes` in the page is the title of the list block, so at this place it makes sense to use plural.
What do you think?

Reply:
By the way, yes it's a menu entry in Design category

Reply:
Let's keep _Email Theme_ in _Admin.Navigation.Menu_.

Reply:
Ok